resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -478,15 +479,36 @@ public DnsNameResolver(
             @Override
             protected void initChannel(DatagramChannel ch) {
                 ch.pipeline().addLast(DATAGRAM_ENCODER, DATAGRAM_DECODER, responseHandler);
+                ch.closeFuture().addListener((ChannelFutureListener) future1 -> {
```

Comment:
nit: use a meaningful name lilke `closeFuture` or since the lambda is not using this variable, just use `__`